### PR TITLE
fix: Allow to use `PlatformColor` in the theme

### PR DESCRIPTION
**Pull request by retyui**
Fix: https://github.com/react-navigation/react-navigation/issues/11481

### CODE REVIEW

File: packages/native-stack/package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "react": "18.2.0",
     "react-native": "0.71.8",
     "react-native-builder-bob": "^0.21.0",
-    "react-native-screens": "~3.20.0",
+    "react-native-screens": "~3.25.0",
```
Comment (retyui) [inline packages/native-stack/package.json:56]: need to use latest ver. with a types update: https://github.com/software-mansion/react-native-screens/pull/1846#issuecomment-1700507895

**Review by osdnk — APPROVED** [review]
Hi,
Thanks for your PR.  Please, next time provide a deeper description of your changes. I didn't understand initially how that change basically "propagates" from rn-screens :)

**Comment by retyui** [comment]
here is more deeper description  https://github.com/react-navigation/react-navigation/pull/11528
but PR hasn't merged yet

**Comment by osdnk** [comment]
Hey, @retyui 

I'm sorry I have to revert this PR #11576.

The concern, which I didn't notice earlier is that as some places (at example/src/Screens/NativeStackHeaderCustomization.tsx) you need to down-cast PlatformColor. Let me think about that. I wonder if this is the only place, but eventually, we shouldn't ask users to do it.

**Comment by retyui** [comment]
right in time when I did a [tweet](https://twitter.com/tell_me_mur/status/1698272773939007627?s=20) lol) 

@osdnk  you don't need to worry about `example/` as theme isn't use a `PlatformColor`'s  there. (Otherwise, you must first resolve this problem: https://github.com/callstack/react-native-paper/issues/4012)

**Comment by osdnk** [comment]
Hi @retyui 
I'm sorry for that, I can empathise it might be disappointing. Let me explain my reasoning here: 
1. It's not about, where is the bug, but more what's the final user experience. With this change in, we get a breaking change (with respect to typings). 
2. In react-navigation, we do color manipulation in many places to improve DX by automating some stuff (search for `Color()` and it only works with string. by supporting `PlatformColor` , those color manipulation are essentially pointless.

Is there a way we can support those kind of manipulations? Also, why do you need this change? Maybe understanding this motivation would help us to find out a good solution.

**Comment by khagesh** [comment]
From our use case, we are trying to use `PlatformColor` for switching between light & dark mode. `PlatformColor` is nice because we don't need to add `if...else` everywhere for theme and then set colors.

The way we can support the manipulation using Color is by checking `PlatformColor` type and then using `resolveColorSync` from `@klarna/platform-colors`. In this way, it won't be a breaking change, and we can have `PlatformColor` as well. But it does add dependency on `@klarna/platform-colors`, and also add checks for `PlatformColor` type inside react-navigation codebase. Personally, I don't think these are huge issues for react-navigation, but you would know better.

**Comment by khagesh** [comment]
I did this workaround for the issue to use `PlatformColor` with react-navigation theme.  Basically I did conversion to string before passing to `NavigationContainer`, and then made sure that NavigationContainer re-renders when color scheme on device changes using `useColorScheme` hook in App.tsx where we render `NavigationContainer`.

```typescript
import { resolveColorSync } from "@klarna/platform-colors";
......

const scheme = useColorScheme();
  const theme = useMemo(() => {
    const navigationThemeColors = {
      primary: resolveColorSync(Colors.primary),
      background: resolveColorSync(Colors.background),
      card: resolveColorSync(Colors.background),
      text: resolveColorSync(Colors.primary),
      border: resolveColorSync(Colors.background),
      notification: resolveColorSync(Colors.primary),
    };
    return {
      ...(scheme === "dark" ? DarkTheme : DefaultTheme),
      colors: {
        ...(scheme === "dark" ? DarkTheme.colors : DefaultTheme.colors),
        ...navigationThemeColors,
      }
    };
  }, [scheme]);

...........

// pass theme 
<NavigationContainer theme={theme}>

```

**Comment by osdnk** [comment]
Alternatively, if you don't rely on our colors' manipulations, you can simply ignore TS error

**Pull request merged**